src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -98,6 +152,13 @@ public void DisposeContext(Context context, Exception exception)
             // If startTimestamp is 0, don't call GetTimestamp, likely don't need the value
             var currentTimestamp = (startTimestamp != 0) ? Stopwatch.GetTimestamp() : 0;
 
+#if NETSTANDARD2_0
+            // TODO: Ability to set duration directly or set end time from timestamp
+            context.Activity?
```

Comment:
End event must be sent before Activity is stopped: consumers will receive End event and use Activity.Current props to log it.

For the same reason, Exception event must also be sent before activity is stopped